src/compiler/program.ts

```diff
@@ -1624,6 +1624,9 @@ namespace ts {
                     collectDynamicImportOrRequireCalls(node);
                 }
             }
+            if ((file.flags & NodeFlags.PossiblyContainsDynamicImport) || isJavaScriptFile) {

```

Comment:
should be `&&` instead of `||`. in a .ts file this is going to be a node, and not a comment.

Reply:
It's always a node. it just may have `.jsDoc` nodes attached.

Reply:
@mhegazy You say that, and I've said that, but that's not true. We parse and bind `jsdoc` in TS files for the LS - we just don't consider them as part of the types of any associated nodes. Seriously - `addJSDocComment` in `parser.ts` is unconditional.